File: Notificare.Core/Models/NotificareDevice.cs
namespace NotificareSdk.Core.Models;

public class NotificareDevice
{
    public string Id { get; }
    public string? UserId { get; }
    public string? UserName { get; }
    public double TimeZoneOffset { get; }
    public NotificareDoNotDisturb? Dnd { get; }
    public IDictionary<string, string> UserData { get; }

    public NotificareDevice(string id, string? userId, string? userName, double timeZoneOffset, NotificareDoNotDisturb? dnd, IDictionary<string, string> userData)
    {
        Id = id;
        UserId = userId;
        UserName = userName;
        TimeZoneOffset = timeZoneOffset;
        Dnd = dnd;
        UserData = userData;
    }
}

File: Notificare.iOS.Binding.Bridge/Sources/Models/NotificareDevice.swift
import Foundation
import NotificareKit

@objc
public class NotificareDevice : NSObject {
    @objc public let deviceId: String
    @objc public let userId: String?
    @objc public let userName: String?
    @objc public let timeZoneOffset: Float
    @objc public let dnd: NotificareDoNotDisturb?
    @objc public let userData: [String : String]
    @objc public let backgroundAppRefresh: Bool

    @objc public init(deviceId: String, userId: String?, userName: String?, timeZoneOffset: Float, dnd: NotificareDoNotDisturb?, userData: [String : String], backgroundAppRefresh: Bool) {
        self.deviceId = deviceId
        self.userId = userId
        self.userName = userName
        self.timeZoneOffset = timeZoneOffset
        self.dnd = dnd
        self.userData = userData
        self.backgroundAppRefresh = backgroundAppRefresh
    }

    public convenience init(from device: NotificareKit.NotificareDevice) {
        self.init(
            deviceId: device.id,
            userId: device.userId,
            userName: device.userName,
            timeZoneOffset: device.timeZoneOffset,
            dnd: device.dnd.map { NotificareDoNotDisturb(from: $0) },
            userData: device.userData,
            backgroundAppRefresh: device.backgroundAppRefresh
        )
    }
}
